File: account-service/internal/model/subscription.go
package model

import "time"

type Subscription struct {
	ID            int64     `json:"id"`
	UserID        int64     `json:"user_id"`
	TierLevel     int       `json:"tier_level"`
	StartTime     time.Time `json:"start_time"`
	EndTime       time.Time `json:"end_time"`
	Status        string    `json:"status"`
	Price         float64   `json:"price"`
	PaymentMethod string    `json:"payment_method,omitempty"`
	OrderID       string    `json:"order_id,omitempty"`
	CreatedAt     time.Time `json:"created_at"`
	UpdatedAt     time.Time `json:"updated_at"`
}

type PurchaseRequest struct {
	UserID        string  `json:"user_id" binding:"required"`
	TierLevel     int     `json:"tier_level" binding:"required,oneof=2 3 4"`
	Price         float64 `json:"price" binding:"required"`
	PaymentMethod string  `json:"payment_method"`
}

type UpgradeRequest struct {
	UserID        string  `json:"user_id" binding:"required"`
	NewTier       int     `json:"new_tier" binding:"required,oneof=3 4"`
	PriceDiff     float64 `json:"price_diff" binding:"required"`
	PaymentMethod string  `json:"payment_method"`
}

type RenewRequest struct {
	UserID        string `json:"user_id" binding:"required"`
	PaymentMethod string `json:"payment_method"`
}

File: ios/AccountCenter/Models/Subscription.swift
import Foundation

struct Subscription: Codable, Identifiable {
    let id: Int64
    let userId: Int64
    let tierLevel: Int
    let startTime: String
    let endTime: String
    let status: String
    let price: Double
    let paymentMethod: String?
    let orderId: String?

    enum CodingKeys: String, CodingKey {
        case id
        case userId = "user_id"
        case tierLevel = "tier_level"
        case startTime = "start_time"
        case endTime = "end_time"
        case status
        case price
        case paymentMethod = "payment_method"
        case orderId = "order_id"
    }
}

struct TierInfo: Codable {
    let userId: Int64
    let identityTier: Int

    enum CodingKeys: String, CodingKey {
        case userId = "user_id"
        case identityTier = "identity_tier"
    }
}
